marimo/_config/config.py

```diff
@@ -513,12 +513,9 @@ class ExperimentalConfig(TypedDict, total=False):
     execution_type: ExecutionType
 
 
-if TYPE_CHECKING:
-    # Prefer to accept any dict since feature flags can change frequently
-    # But maintain type safety for known flags
-    ExperimentalConfigType = Union[dict[str, Any], ExperimentalConfig]
-else:
-    ExperimentalConfigType = dict
+# Prefer to accept any dict since feature flags can change frequently
+# But maintain type safety for known flags
+ExperimentalConfigType = dict[str, Any]
```

Comment:
Comment is outdated if we want to get rid of the static checking for the typed dict